src/App.tsx

```
@@ -1,37 +1,62 @@
 import React from 'react';
 import './App.scss';
-
+import { Clock } from './components/Clock';
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">{clockName}</strong>
-
-        {' time is '}
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId?: number;
+
+  handleContextmenu = () => {
+    this.setState({ hasClock: false });
+  };
```

This handler does not call `event.preventDefault()` as required by the task description. The description example shows the `contextmenu` handler must call `event.preventDefault(); // not to show the context menu` before hiding the clock. Update the handler to accept the event and call `event.preventDefault()` before changing state, for example:

`handleContextmenu = (event: MouseEvent) => { event.preventDefault(); this.setState({ hasClock: false }); }`

Reference requirement: "event.preventDefault(); // not to show the context menu" (description).